homeassistant/components/backup/manager.py

```diff
@@ -433,10 +437,12 @@ async def _async_create_backup(
     ) -> Backup:
         """Generate a backup."""
         success = False
+
         if LOCAL_AGENT_ID in agent_ids:
-            backup_dir = self.backup_dir
+            local_agent = cast(LocalBackupAgent, self.backup_agents[LOCAL_AGENT_ID])
+            tar_file_path = local_agent.get_backup_path(slug)
         else:
-            backup_dir = self.temp_backup_dir
+            tar_file_path = self.temp_backup_dir / f"{slug}.tar"
```

Comment:
I'd suggest we instead always write to the backups directory and let the builtin agent do a no-op in its upload method.